### DISCUSSION TITLE
LDM Notes for November 30th, 2022

### DISCUSSION THREAD

**Discussion by 333fred** [dotnet/csharplang#6766] [LDM Notes]
https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-11-30.md

## Agenda

- Revise `MemberNotNull` for `required`
- `params Span<T>`

**Comment by RikkiGibson** [comment]
> Should users be able to turn off the feature at the compilation-level such that backing arrays are always heap allocated?

I would expect that `params ReadOnlySpan<T>` methods will not be invokable in expanded form before the C# version the feature releases in. Is that right?

Maybe in some cases people will be able to *turn off the feature* for triage purposes by downgrading language version (not TargetFramework), and making their code go back to binding to the array overload.

**Reply by 333fred** [reply]
I don't think that downgrading langversion is a viable testing switch here.

**Reply by sab39** [reply]
What would be the value of turning it off at the compilation level? I could see wanting to turn it off on a case-by-case basis, potentially, but turning it off globally would affect calls to BCL methods, calls to methods in the code that's being compiled, and calls to methods in third-party libraries that use the feature. Even in the scenario described, where the stack allocation is causing problems, not _every_ use of the feature will cause problems, and switching _all_ of them to heap allocation would surely change the performance characteristics of the program in other ways, obscuring the very thing we're trying to make measurable.

Switching it _for calls to a particular method_ might be more helpful in that testing scenario, but that sounds too specialized and niche to be worth the engineering effort.
